src/utils/sentry/httpclient.ts

```diff
@@ -0,0 +1,486 @@
+        let cookies
+
+        try {
+          const cookieString =
+            headers[cookieHeader] || headers[cookieHeader.toLowerCase()] || undefined
+
+          if (cookieString) {
+            cookies = _parseCookieString(cookieString)
+          }
+        } catch (e) {
+          logger.log(`Could not extract cookies from header ${cookieHeader}`)
+        }
+
+        return {
+          headers,
+          cookies,
+        }
+      })
+    }
+
+    const event = _createEvent({
+      request: {
+        url: request.url,
+        method: request.method,
+        requestHeaders,
+        requestCookies,
+        data: requestInit?.body,
+        query_string: request.url.includes('?') ? request.url.split('?')[1] : undefined,
+      },
+      response: {
+        status: response.status,
+        responseHeaders,
+        responseCookies,
+      },
+    })
+
+    captureEvent(event)
+  }
+}
+
+/**
+ * Interceptor function for XHR requests
+ *
+ * @param xhr The XHR request
+ * @param method The HTTP method
+ * @param headers The HTTP headers
+ */
+function _xhrResponseHandler(
+  options: HttpClientOptions,
+  xhr: SentryWrappedXMLHttpRequest & XMLHttpRequest,
+  method: string,
+  headers: Record<string, string>,
+): void {
+  const shouldCapture = _shouldCaptureResponse(options, xhr.status, xhr.responseURL)
+
+  if (shouldCapture) {
+    let requestHeaders, responseCookies, responseHeaders
+
+    if (_shouldSendDefaultPii()) {
+      try {
+        const cookieString =
+          xhr.getResponseHeader('Set-Cookie') || xhr.getResponseHeader('set-cookie') || undefined
+
+        if (cookieString) {
+          responseCookies = _parseCookieString(cookieString)
+        }
+      } catch (e) {
+        logger.log('Could not extract cookies from response headers')
+      }
+
+      try {
+        responseHeaders = _getXHRResponseHeaders(xhr)
+      } catch (e) {
+        logger.log('Could not extract headers from response')
+      }
+
+      requestHeaders = headers
+    }
+
+    const event = _createEvent({
+      request: {
+        url: xhr.responseURL,
+        method,
+        requestHeaders,
+        query_string: xhr.__sentry_xhr_v3__?.url.includes('?')
+          ? xhr.__sentry_xhr_v3__.url.split('?')[1]
+          : undefined,
+        data: xhr.__sentry_xhr_v3__?.body,
+      },
+      response: {
+        // Can't access request cookies from XHR
+        responseHeaders,
+        responseCookies,
+        status: xhr.status,
+      },
+    })
+
+    captureEvent(event)
+  }
+}
+
+/**
+ * Extracts response size from `Content-Length` header when possible
+ *
+ * @param headers
+ * @returns The response size in bytes or undefined
+ */
+function _getResponseSizeFromHeaders(headers?: Record<string, string>): number | undefined {
+  if (headers) {
+    const contentLength = headers['Content-Length'] || headers['content-length']
+
+    if (contentLength) {
+      return parseInt(contentLength, 10)
```

Comment:
Nice marking radix as 10, per the docs:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt#radix

> If 0 or not provided, the radix will be inferred based on string's value. Be careful — this does not always default to 10

Reply:
But why even make it optional JS?! if the behavior is not clearly defined 💀